sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -313,4 +313,13 @@ export class InstitutionService extends RecordDataModelService<Institution> {
         .map((roleObject) => roleObject.role),
     };
   }
+
+  async removeUser(institutionUser: InstitutionUser) {
+    await this.institutionUserRepo.remove(institutionUser);
```

Comment:
Please consider changing both queries to be executed in Parallel.
Please considering decorating the method with a transaction.